src/WiFiClient.cpp

```diff
@@ -152,6 +152,8 @@ int WiFiClient::connect(IPAddress ip, uint16_t port, uint8_t opt, const uint8_t
 		_socket = -1;
 		return 0;
 	}
+
+	WiFi._client[_socket] = this;
```

Comment:
I'm wondering if we should move this down just above the `return 1;` statement at the end.

What do you think?